workspace/lib/lita/handlers/teamcity.rb

```diff
@@ -615,6 +649,88 @@ def format_artifacts_build(build_type, build_id, build_number)
       def is_numeric(number)
         true if Integer(number) rescue false
       end
+
+      def fetch_download_url_by_pr_id(pr_id)
+        site_info = JSON.parse(config.artifactory_info)
+        site_url = site_info['site_url']
+        pr_path = site_info['pr_path']
+        full_url = "#{site_url}api/storage/#{pr_path}#{pr_id}"
+        specific_url = "#{pr_id}"
+        temp_uri = fetch_url_of_build(full_url)
+        while temp_uri != nil
+          full_url += temp_uri
+          specific_url += temp_uri
```

Comment:
similar to **full_url** then I think we need a common variable for this value, append it to them when finishing.